File: stepik-java-api/src/main/java/org/stepik/api/actions/StepikAttemptsAction.java
package org.stepik.api.actions;

import org.jetbrains.annotations.NotNull;
import org.stepik.api.client.StepikApiClient;
import org.stepik.api.objects.ObjectsContainer;
import org.stepik.api.queries.attempts.StepikAttemptsGetQuery;
import org.stepik.api.queries.attempts.StepikAttemptsPostQuery;

/**
 * @author meanmail
 */
public class StepikAttemptsAction extends StepikAbstractAction {

    public StepikAttemptsAction(@NotNull StepikApiClient stepikApiClient) {
        super(stepikApiClient);
    }

    @NotNull
    public <R extends ObjectsContainer> StepikAttemptsPostQuery post() {
        return new StepikAttemptsPostQuery(this);
    }

    public <R extends ObjectsContainer> StepikAttemptsGetQuery get() {
        return new StepikAttemptsGetQuery(this);
    }
}

File: stepik-java-api/src/main/kotlin/org/stepik/api/actions/StepikAttemptsAction.kt
package org.stepik.api.actions

import org.stepik.api.client.StepikApiClient
import org.stepik.api.queries.attempts.StepikAttemptsGetQuery
import org.stepik.api.queries.attempts.StepikAttemptsPostQuery

class StepikAttemptsAction(stepikApiClient: StepikApiClient) : StepikAbstractAction(stepikApiClient) {
    
    fun post(): StepikAttemptsPostQuery {
        return StepikAttemptsPostQuery(this)
    }
    
    fun get(): StepikAttemptsGetQuery {
        return StepikAttemptsGetQuery(this)
    }
}
